### PR TITLE
🏗️ maintenance week-9: upgrades test&tool requirements

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Upgrades ONLY dependencies needed for testing and tooling in the entire repository.  Notice that in the case of ``packages`` modules, it also include upgrades in ``base.txt`` because in reality those requirements are used ONLY for testing. All modules there are libraries that are tested against latest releases of its dependencies but adds them as "unfrozen" requirements (see ``setup.py`` in every package).

Service requirement upgrades will be done in a separate PR

### Upgrade highlights:

- pylint 2.6.0 -> 2.7.2. - [changelog](http://pylint.pycqa.org/en/2.7.1/whatsnew/2.7.html). Some new linter error detected 
    - [inconsistent-return-statements](https://pycodequ.al/docs/pylint-messages/r1710-inconsistent-return-statements.html)
    - consider-using-generator
- pydantic  1.7.3->1.8.1 - [changelog](https://pydantic-docs.helpmanual.io/changelog/)
    - @sanderegg please review carefuly changes I did in ``simcore_sdk``

## Related issue/s

weekly maintenance

## How to test

all CI tests shall pass

## Checklist

- [x] upgrade reqs for testing&tooling

### CODE REVIEW

File: packages/models-library/src/models_library/services.py [hunk 1]
```diff
@@ -180,11 +180,6 @@ class ServiceProperty(BaseModel):
         None, description="Units, when it refers to a physical quantity"
     )
 
-    # TODO: use discriminators
-    unit: Optional[str] = Field(
-        None, description="Units, when it refers to a physical quantity"
-    )
-
```
Comment (GitHK) [inline packages/models-library/src/models_library/services.py:187]: ahh this was duplicated

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py [hunk 2]
```diff
@@ -3,14 +3,21 @@
...
 from models_library.utils.nodes import compute_node_hash
+from packaging import version
 
 from ..node_ports.dbmanager import DBManager
 from ..node_ports.exceptions import InvalidProtocolError
 from .nodeports_v2 import Nodeports
 
+# NOTE: Keeps backwards compatibility with pydantic
+#   - from 1.8, it DOES NOT need __root__ specifiers when nested
+_PYDANTIC_NEEDS_ROOT_SPECIFIED = version.parse(pydantic.VERSION) < version.parse("1.8")
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py:18]: Why youse this? When you install simcore_sdk, you will require pydantic's version inside this package and result in collisions.

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/ports_mapping.py [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+from typing import Dict, ItemsView, Iterator, KeysView, Type, Union, ValuesView
 
 from models_library.services import PROPERTY_KEY_RE
 from pydantic import BaseModel, constr
 
 from ..node_ports.exceptions import UnboundPortError
 from .port import Port
 
-PortKey: constr = constr(regex=PROPERTY_KEY_RE)
+PortKey: Type[str] = constr(regex=PROPERTY_KEY_RE)
```
Comment (GitHK) [inline packages/simcore-sdk/src/simcore_sdk/node_ports_v2/ports_mapping.py:9]: I do not understand this change here

**Review by GitHK — COMMENTED** [review]
Please see below

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/ports_mapping.py (hunk 3 above)
Comment (pcrespov) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/ports_mapping.py:9]: - you use types for annotations
- ``constr`` is a function-factory that creates constraint string **types** but not a type itself
If still unclear, look at pydantic code

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py (hunk 2 above)
Comment (pcrespov) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py:18]: - pydantic > 1.8 has some breaking changes.
- simcore-sdk is a library that requires pydantic (i.e. we do not impose strong constraints)
- services installing ``simcore-sdk`` determine which version to use
- simcore-sdk keeps backwards compatibility by including two code paths depending on the version of pydantic
- all services are happy

I do not understand what you mean by "collisions" ...

**Review by mguidon — APPROVED** [review]
Nothing bad jumps to mind

File: packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py (hunk 2 above)
Comment (GitHK) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py:18]: When you install `simcore-sdk` you require these [packages/simcore-sdk/requirements/_base.txt#L108](https://github.com/ITISFoundation/osparc-simcore/blob/33e68f7912976603343fdfed855928af1db84276/packages/simcore-sdk/requirements/_base.txt#L108) That's what I mean. You already have 1.8 installed. What am I missing here?

Comment (pcrespov) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py:18]: @GitHK : following your comment.

 All ``*.txt``-like requirements folder in any project UNDER ``packages`` are effectively **just to test against "the latest versions of everything"** (more details in ``/home/crespo/devp/osparc-simcore/requirements/python-dependencies.md``) ... but the real package "requirements" are in ``_base.in``. 

When we include this package in a service, the contraints on the service are determined by ``_base.in`` ... therefore there is no-collision. Have a look at [web/server/requirements.in](https://github.com/ITISFoundation/osparc-simcore/blob/fdd28e2d7edd9ae65955dbc23538f81ce584ba25/services/web/server/requirements/_base.in#L5)

TIP: To understand better the latter, open ``packages/service-library/setup.py`` and see how ``install_requires`` attribute is filled.

Comment (GitHK) [reply packages/simcore-sdk/src/simcore_sdk/node_ports_v2/serialization_v2.py:18]: Perfect, I've missed that one. It is now clear to me.

**Review by GitHK — APPROVED** [review]

**Pull request merged**